### PR TITLE
Refactor to separate .appengine.compat from other bundles

### PR DESCRIPTION
**Pull request by briandealwis**
Merge .`.compat.cte13` into`.compat`. Removes dependency of `.appengine.facets` on `.appengine.compat`. Copy `AppEngineStandardProjectConverter*` into a corresponding `GpeConvert*`.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertJob.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertJob.java]: copyright

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java]: copyright

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          true /* convert to faceted project if necessary */, null /* no monitor here */);
+      if (AppEngineStandardFacet.hasFacet(facetedProject)) {
+        throw new IllegalStateException("Convert menu enabled for App Engine projects");
+      }
+
+      GpeConvertJob job = new GpeConvertJob(facetedProject);
+      job.setUser(true);
+      job.schedule();
+    } catch (CoreException ex) {
+      ErrorDialog.openError(shell, "Conversion Failed", "Failed to convert to a faceted project",
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java]: Is StatusManager preferred for this nowadays?

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java]: Weird. My settings are configured to produce the header, and it works sometimes.

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java (hunk 3 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java]: You're right; SM also updates as further errors are accumulated.

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java [hunk 4]
```diff
@@ -42,8 +58,8 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
       job.setUser(true);
       job.schedule();
     } catch (CoreException ex) {
-      ErrorDialog.openError(shell, "Conversion Failed", "Failed to convert to a faceted project",
-          ex.getStatus());
+      StatusManager.getManager()
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java]: StatusUtil.setErrorStatus will do this part for you.

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/AppEngineStandardProjectConvertCommandHandler.java [hunk 5]
```diff
@@ -59,8 +60,8 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
       job.setUser(true);
       job.schedule();
     } catch (CoreException ex) {
-      ErrorDialog.openError(shell, "Conversion Failed", "Failed to convert to a faceted project",
-          ex.getStatus());
+      StatusManager.getManager()
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/AppEngineStandardProjectConvertCommandHandler.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertJob.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+
+public class GpeConvertJob extends Job {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertJob.java]: I wonder if we can extend `AppEngineStandardProjectConvertJob`. After all, you want and have to do precisely what `AppEngineStandardProjectConvertJob` does, after removing GPE stuff. Not only removes code dup, safe when we add more functions to `AppEngineStandardProjectConvertJob`.

**Comment by chanseokoh** [comment]
We usually put a single `Message.java` and `messages.properties` per bundle at the base package. `c.g.c.t.e.a.compat` in this case.

**Comment by briandealwis** [comment]
The use of `isAccessible()` also avoids logged exceptions from attempting to access closed projects

```
Oct 30, 2017 2:12:37 PM com.google.cloud.tools.eclipse.appengine.compat.cte13.CloudToolsEclipseProjectUpdater hasOldContainers
WARNING: Skipping project: verify-feature-completeness
org.eclipse.core.internal.resources.ResourceException(/verify-feature-completeness)[372]: java.lang.Exception: Project 'verify-feature-completeness' is not open.
	at org.eclipse.core.internal.resources.ResourceException.provideStackTrace(ResourceException.java:39)
	at org.eclipse.core.internal.resources.ResourceException.<init>(ResourceException.java:35)
	at org.eclipse.core.internal.resources.Project.checkAccessible(Project.java:144)
	at org.eclipse.core.internal.resources.Project.hasNature(Project.java:531)
	at com.google.cloud.tools.eclipse.appengine.compat.cte13.CloudToolsEclipseProjectUpdater.hasOldContainers(CloudToolsEclipseProjectUpdater.java:52)
	at com.google.cloud.tools.eclipse.appengine.compat.cte13.CloudToolsEclipseProjectNotifier.findCandidates(CloudToolsEclipseProjectNotifier.java:87)
	at com.google.cloud.tools.eclipse.appengine.compat.cte13.CloudToolsEclipseProjectNotifier.access$0(CloudToolsEclipseProjectNotifier.java:81)
	at com.google.cloud.tools.eclipse.appengine.compat.cte13.CloudToolsEclipseProjectNotifier$1.runInWorkspace(CloudToolsEclipseProjectNotifier.java:61)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
```

**Pull request merged**